easybuild/easyblocks/l/llvm.py

```diff
@@ -408,28 +409,56 @@ def __init__(self, *args, **kwargs):
         if exp_targets:
             self.log.warning("Experimental targets %s are being used.", ', '.join(exp_targets))
 
-        self.build_targets = build_targets or []
+        all_target_cond = 'all' in build_targets
+        self.nvptx_target_cond = (BUILD_TARGET_NVPTX in build_targets) or all_target_cond
+        self.amdgpu_target_cond = (BUILD_TARGET_AMDGPU in build_targets) or all_target_cond
+
+        if ('cuda' in self.deps or cuda_toolchain) and not self.nvptx_target_cond:
+            raise EasyBuildError("CUDA dependency detected, but NVPTX not in manually specified build targets")
+        if cuda_cc_list and not self.nvptx_target_cond:
+            raise EasyBuildError(
+                "CUDA compute capabilities specified, but NVPTX not in manually specified build targets"
+            )
+        if 'rocr-runtime' in self.deps and not self.amdgpu_target_cond:
+            raise EasyBuildError(
+                "ROCR-Runtime dependency detected, but AMDGPU not in manually specified build targets"
+            )
+        if amd_gfx_list and not self.amdgpu_target_cond:
+            raise EasyBuildError("AMD GPU list specified, but AMDGPU not in manually specified build targets")
 
-        self.cuda_cc = [cc.replace('.', '') for cc in cuda_cc_list]
-        if BUILD_TARGET_NVPTX in self.build_targets and not self.cuda_cc:
-            raise EasyBuildError("Can't build Clang with CUDA support without specifying 'cuda-compute-capabilities'")
-        if self.cuda_cc and BUILD_TARGET_NVPTX not in self.build_targets:
-            print_warning("CUDA compute capabilities specified, but NVPTX not in manually specified build targets.")
+        self.build_targets = build_targets or []
 
-        self.amd_gfx = amd_gfx_list
-        if BUILD_TARGET_AMDGPU in self.build_targets and not self.amd_gfx:
-            raise EasyBuildError("Can't build Clang with AMDGPU support without specifying 'amd_gfx_list'")
-        if self.amd_gfx and BUILD_TARGET_AMDGPU not in self.build_targets:
-            print_warning("'amd_gfx' specified, but AMDGPU not in manually specified build targets.")
+        # Enable offload targets for LLVM >= 18
+        self.cuda_cc = []
+        self.amd_gfx = []
+        if self.cfg['build_openmp_offload'] and LooseVersion(self.version) >= LooseVersion('18'):
+            if self.nvptx_target_cond:
+                if LooseVersion(self.version) < LooseVersion('20') and not cuda_cc_list:
+                    raise EasyBuildError(
+                        f"LLVM < 20 requires 'cuda-compute-capabilities' to build with {BUILD_TARGET_NVPTX}"
```

Comment:
Try to be consistent with AMD below
```suggestion
                        f"LLVM < 20 requires 'cuda_compute_capabilities' to build with {BUILD_TARGET_NVPTX}"
```